driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -295,47 +329,22 @@ public CodecRegistry register(TypeCodec<?>... codecs) {
      * @return this Builder (for method chaining).
      */
     public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs) {
-        return register(codecs, false);
-    }
-
-    /**
-     * Register the given codecs with this registry.
-     *
-     * @param codecs The codecs to add to the registry.
-     * @param includeDerivedCollectionCodecs If {@code true}, register not only the given codecs
-     * but also collection codecs whose element types are handled by the given codecs, i.e. for
-     * each codec:
-     * <ul>
-     *   <li>lists of the codec's type;</li>
-     *   <li>sets of the codec's type;</li>
-     *   <li>maps of a primitive type to the codec's type, or of the codec's type to a primitive
-     *   type;</li>
-     * </ul>
-     * @return this Builder (for method chaining).
-     */
-    public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs, boolean includeDerivedCollectionCodecs) {
-        if(immutable)
-            throw new UnsupportedOperationException("This CodecRegistry instance is immutable");
         for (TypeCodec<?> codec : codecs) {
             // add this codec to the beginning of the list so that it gets higher priority
             this.codecs.add(codec);
-            if(includeDerivedCollectionCodecs){
```

Comment:
Including derived collections is probably a premature optimization; it only grows the internal list which makes it less and less efficient. And codecs for collections of a basic type can be created on the fly anyway.